plugins/@grouparoo/customerio/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,88 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+
+  const customerId = newProfileProperties["customer_id"];
+  const oldCustomerId = oldProfileProperties["customer_id"];
+
+  if (!customerId) {
+    throw new Error(`newProfileProperties[customer_id] is a required mapping`);
+  }
+
+  if (oldCustomerId !== undefined && customerId !== oldCustomerId) {
+    // Must delete old customer if ID has changed
+    await client.destroy(oldCustomerId);
+  }
+
+  if (toDelete) {
+    await client.destroy(customerId);
+    return { success: true };
+  }
+
+  const payload: any = {};
+
+  // set profile properties, including old ones
+  const newKeys = Object.keys(newProfileProperties);
+  const oldKeys = Object.keys(oldProfileProperties);
+  const allKeys = oldKeys.concat(newKeys);
```

Comment:
For clarity, Might be useful to do the `new Set` thing here so there are not duplicates.